mantidproject/mantid
# ISIS Powder PEARL - revert instrument settings to default after every method call

The way the PEARL object handles its settings was changed. Now, instead of every method call updating the instrument settings with its parameters, any settings given in the constructor are taken as defaults, and settings changed in a method call (ie `focus`, `create_vanadium` or `create_cal`) are only applied during execution of that method. On termination, the default settings are restored

**To test:**

If not reviewing from ISIS, you'll need to change the second line and do some setup locally - ping me a message for an explanation of how to set up ISIS_Powder directories, as it can be fiddly.

```
from isis_powder import Pearl

pearl = Pearl(config_file=r"\\olympic\Babylon5\Public\JoeRamsay\PEARL\pearl_shared_config.yaml")

pearl.focus(run_number="99504")
pearl.focus(run_number="99506", tt_mode="tt88") # Should show an warning about replacing this parameter
pearl.focus(run_number="99507")

print(pearl._inst_settings.tt_mode)
```

You should get 3 GroupWorkspaces - `PEARL99504_tt70-Results-D-Grp`, `PEARL99506_tt88-Results-D-Grp` and `PEARL99507_tt70-Results-D-Grp`. The final line should print `tt88`.

Fiddle with any parameters you like, and make sure that any that you change are reverted back to the defaults after that line has finished. Also try running likes in non-sequential order (by highlighting and `Ctrl-Enter`) - make sure that the correct settings are still used

Fixes #22280 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/compare/22280_isis_powder_pearl_default_settings?expand=1#diff-b2bd2540c595f643521322a749fbd5dc)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
